src/main/fc/fc_msp.c

```diff
@@ -1987,15 +2001,39 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
     switch (cmdMSP) {
 #ifdef TRANSPONDER
-    case MSP_SET_TRANSPONDER_CONFIG:
-        if (dataSize != sizeof(transponderConfig()->data)) {
+    case MSP_SET_TRANSPONDER_CONFIG: {
```

Comment:
Same here: `dataSize == sizeof(transponderConfig()->data` => it's ILap data, otherwise read provider info.